### ISSUE TITLE
breaks on ILogger when used inside an AbstractAcceptanceTest

### ISSUE THREAD

**Issue by vip32** [JamesRandall/FunctionMonkey#69]
```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=FunctionMonkey
  StackTrace:
   at FunctionMonkey.Infrastructure.FunctionLogger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, EventId eventId, Exception exception, String message, Object[] args)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, String message, Object[] args)
   at Vpp.DynamicPricing.Application.GetEchoQueryHandler.ExecuteAsync(GetEchoQuery command) in D:\_projects\Vpp.DynamicPricing.ssh\src\Application\Queries\GetEchoQueryHandler.cs:line 43
   at AzureFromTheTrenches.Commanding.Implementation.CommandHandlerExecuter.<ExecuteActorForCommandWithNoResult>d__4.MoveNext()
```

my test:
```
public class GetEchoQueryTests : AbstractAcceptanceTest
    {
        [Fact]
        public async Task ReturnOkResultForQuery()
        {
            var response = await this.ExecuteHttpAsync(
                new GetEchoQuery(),
                HttpMethod.Get);

            response.StatusCode.ShouldBe(200);
        }
    }
```
where the handler accepts an **ILogger** in the ctor. 

Workaround: inject an **ILoggerFactory** and create an ILogger from that in the ctor, like this:
```
public class GetEchoQueryHandler : ICommandHandler<GetEchoQuery>
    {
        private readonly ILogger logger;

        public GetEchoQueryHandler(ILoggerFactory loggerFactory)
        {
             this.logger = loggerFactory.CreateLogger<GetEchoQueryHandler>();
        }
```

**Comment by JamesRandall** [comment]
Hmm. Interesting. I'll take a look at that - I'm writing some acceptance tests at the moment. I think I know what is going on and it should be (famous last words) a simple fix.

Thanks for reporting.

**Comment by JamesRandall** [comment]
I've resolved this so that ILogger is now accessible, you can also inject a mock logger if you want to check critical log output for example.

**Comment by JamesRandall** [comment]
Addressed in v3.0.4-beta1. Thanks!
